code/modules/projectiles/gun_helpers.dm

```diff
@@ -565,24 +574,27 @@ should be alright.
 		if(GUN_FIREMODE_AUTOBURST)
 			new_firemode = GUN_FIREMODE_BURSTFIRE
 	if(!(new_firemode in gun_firemode_list))
-		to_chat(user, "<span class='warning'>[src] lacks a [new_firemode]!</span>")
+		to_chat(usr, "<span class='warning'>[src] lacks a [new_firemode]!</span>")
 		return
-	do_toggle_firemode(user, new_firemode)
+	do_toggle_firemode(usr, new_firemode)
 
 
-/obj/item/weapon/gun/verb/toggle_burstfire()
+/mob/living/carbon/human/verb/toggle_burstfire()
 	set category = "Weapons"
 	set name = "Toggle Burst Fire"
 	set desc = "Toggle burst firemode, if the gun has it."
-	set src = usr.contents
 
-	var/obj/item/weapon/gun/automatic_gun = get_active_firearm(usr)
-	if(!automatic_gun)
+	var/obj/item/weapon/gun/G = get_active_firearm(usr)
+	if(!G)
 		return
-	automatic_gun.do_toggle_burst(usr)
+	G.toggle_burstfire()
 
 
-/obj/item/weapon/gun/proc/do_toggle_burst(mob/user)
+/obj/item/weapon/gun/verb/toggle_burstfire()
+	set category = null
```

Comment:
Does category not default to null?
Ditto for the rest of the verbs

Reply:
Right. I thought it would fix the double usage message if verbs for different objects have the same name.

Reply:
Done.